torch/utils/data/graph.py

```diff
@@ -74,6 +79,18 @@ def reduce_hook(obj):
 
 
 def traverse(datapipe: DataPipe, only_datapipe: bool = False) -> DataPipeGraph:
+    r"""
+    Traverse the DataPipes and their attributes to extract the DataPipe graph. When
+    ``only_dataPipe`` is specified as ``True``, it would only look into the attribute
+    from each DataPipe that is either a DataPipe and a Python collection object such as
+    ``list``, ``tuple``, ``set`` and ``dict``.
+    Args:
+        datapipe: the end DataPipe of the graph
+        only_datapipe: If ``False`` (default), all attributes of each DataPipe are traversed
```

Comment:
nit: it seems like the default usages always set `only_datapipe=True`? Can you provide an example when you want it to be `False`? And perhaps mention that example in here for future reference?

Reply:
This is for BC. We previously don't have `only_datapipe` option for `traverse` function. So, previously, this function behaves like `only_datapipe=False`.
I can add a deprecation warning here to indicate users to remove `only_datapipe` argument.

Reply:
Yes I think a deprecation warning will be good if we do not expect it to be used by users in the future.